airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -267,3 +267,14 @@ def _wrapped_primary_key(keys: Optional[Union[str, List[str], List[List[str]]]])
             return wrapped_keys
         else:
             raise ValueError(f"Element must be either list or str. Got: {type(keys)}")
+
+    @property
+    def catalog_schema(self):
+        if hasattr(self, "_catalog_schema"):
+            return self._catalog_schema
+        return None
+
+    # noinspection PyAttributeOutsideInit
+    @catalog_schema.setter
+    def catalog_schema(self, schema: Dict[str, Any]):
```

Comment:
Should this be an object instead of a primitive type? Can we re-use `ConfiguredAirbyteCatalog`?